packages/catalog-search/src/SearchBox.jsx

```diff
@@ -54,40 +66,65 @@ export const SearchBoxBase = ({
     }
   };
 
+  const handleSuggestionSubmit = (hit) => {
+    // If an override is provided, call it with the hit
+    if (typeof suggestionSubmitOverride === 'function') {
+      suggestionSubmitOverride(hit);
+    }
+
+    // When a suggested search is clicked, set the course tile to the search query
+    setSearchQuery(hit.title);
```

Comment:
My only concern now is that after we have split View all click handler, is there any need for these lines?
```
setSearchQuery(hit.title);
handleSubmit(hit.title);
```

because this function will only be called when someone clicks some item from suggestion and redirect is disabled. So this function wont be called on learner portal but only on explore catalog, so do you guys want this behaviour that when user clicks on suggestion then along with you callback getting called, you want all the search results to be refined according to the title of item user just clicked? and searchbox to contain whatever the title of item user clicked? if yes then all good.

Reply:
Hmm, I was trying to preserve the behavior of the original learner portal which is why I excluded those from the handle submit method. The thinking is that there are 3 functionalities currently supported (and @binodpant had some great suggestions as to how we might be able to get ourselves to just two or even one) Currently the search header/box suggestions work in such a way that:
- if  the `disableSuggestionRedirect`prop is provided, then a base behavior of setting the facet/input value to the suggested search title is used instead of redirecting to the course page
- if `suggestionSubmitOverride` is provided along with `disableSuggestionRedirect` then an additional custom behavior is applied to the onclick of the suggestion along with the new base behavior
- if no new props are provided then the behavior reflects current day learner portal implementation (we might rework this to the above `additional behavior` functionality.

(also side note, not giving the index prop will disable suggestions all together)